tests/app/views/test_agreements.py

```diff
@@ -778,3 +778,69 @@ def test_can_not_put_framework_agreement_on_hold_that_has_no_framework_agreement
         assert res.status_code == 400
         error_message = json.loads(res.get_data(as_text=True))['error']
         assert error_message == "Framework agreement must have a 'frameworkAgreementVersion' to be put on hold"
+
+
+class TestTemporaryCountersignRoute(BaseFrameworkAgreementTest):
+    """Test related to temporary counter sign route
+
+    When temporary countersign route is removed, these tests can also be removed
+    """
+    def temp_countersign(self, agreement_id, countersign_path, countersign_returned_at):
+        return self.client.post(
+            '/agreements/{}/countersign-script'.format(agreement_id),
+            data=json.dumps(
+                {
+                    "updated_by": "interested@example.com",
+                    "agreement": {
+                        "countersignedAgreementPath": countersign_path,
+                        "countersignedAgreementReturnedAt": countersign_returned_at
+                    }
+
+                }),
+            content_type='application/json')
+
+    def test_temp_countersign(self, supplier_framework):
+        agreement_id = self.create_agreement(
+            supplier_framework,
+            signed_agreement_returned_at=datetime(2016, 10, 1),
+        )
+
+        res = self.temp_countersign(
+            agreement_id,
+            'example/path/file.pdf',
+            '2016-11-01T00:00:00.000000Z'
```

Comment:
Would it be useful to have a comment on the route about the time format required for it to work properly?